pylint/checkers/base.py

```diff
@@ -1218,6 +1303,32 @@ def open(self):
             for name_type in group.split(':'):
                 self._name_group[name_type] = 'group_%s' % (group,)
 
+        regexps, hints = self._create_naming_rules()
+        self._name_regexps = regexps
+        self._name_hints = hints
+
+    def _create_naming_rules(self):
+        regexps = {}
+        hints = {}
+
+        for name_type in KNOWN_NODE_TYPES:
+            naming_style_option_name = "%s_naming_style" % (name_type,)
+            naming_style_name = getattr(self.config, naming_style_option_name)
+
+            regexps[name_type] = NAMING_STYLES[naming_style_name].get_regex(name_type)
+
+            custom_regex_setting_name = "%s_rgx" % (name_type, )
```

Comment:
> currently we will always pick regexes, due to already having the -rgx hints pre-generated in config files

Yes, and I think it's fine, and should be kept this way. By doing it we won't introduce any regressions to current clients. On the same time, new users will happily use this basic functionality.

We simply need reasonable documentation of new capabilities, both in changelog / what's new and in "naming convention" section in docs.